File: Lecture 5 Bitwise Operators, For Loops, Operator Precedence & Variable Scoping/numberOfOneBits.cpp
class Solution {
public:
    int hammingWeight(int n) {
    
        int count = 0;
        while(n!=0) {
            
            //checking last bit
            if(n&1) {
                count++;
            }
            n = n>>1;
        }
        return count;
    }
};

File: Lecture005 Bitwise Operators, For Loops, Operator Precedence & Variable Scoping/numberOfOneBits.java
class Solution {
    // you need to treat n as an unsigned value
    public int hammingWeight(int n) {

        int count = 0;

        for (int i = 1; i <= 32; i++) {
            if ((n & 1) == 1) {
                count++;
            }
            n = n >> 1; // right shift all the bits by one place
        }

        return count;

    }
}
